test/mock/MockXcmRouter.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.7.6;
+pragma abicoder v2;
+
+import {TypedMemView} from "memview-sol/TypedMemView.sol";
+import "forge-std/Test.sol";
+import {CentrifugeConnector} from "src/Connector.sol";
+import {ConnectorMessages} from "src/Messages.sol";
+
+contract MockXcmRouter is Test {
+    using TypedMemView for bytes;
+    using TypedMemView for bytes29;
+
+    CentrifugeConnector public immutable connector;
+
+    constructor(CentrifugeConnector connector_) {
+        connector = connector_;
+    }
+
+    modifier onlyConnector() {
+        require(msg.sender == address(connector), "ConnectorXCMRouter/only-connector-allowed-to-call");
+        _;
+    }
+
+    function handle(bytes memory _message) external {
```

Comment:
ℹ️ I was first having this `MockXcmRouter` wrap `ConnectorXcmRouter` and just delegate the handle down to the actual xcm router but that would require it being possible to only allow that inner/actual `router` to call `Connector` functions while then having `Connector.transfer` calling `MockXcmRouter.send` instead of `ConnectorXcmRouter.send` (i.e, we would need to have two routers in the `Connector` contract, one for incoming calls and one for the outgoing `send`)